### PR TITLE
feat: Add ability to disable table inline edit for specific cells

### PR DESCRIPTION
**Pull request by timogasda**
### Description

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/table/__integ__/inline-editing.test.ts [hunk 1]
```diff
@@ -29,6 +29,10 @@ const cellBelow$ = tableWrapper.findEditCellButton(5, 5).toSelector();
 
 const bodyCellError = bodyCell.findFormField().findError().toSelector();
 
+const disabledCell = tableWrapper.findBodyCell(4, 4);
+const disabledCell$ = tableWrapper.findBodyCell(4, 4).toSelector();
```
Comment (abdhalees) [inline src/table/__integ__/inline-editing.test.ts:33]: nit
```suggestion
const disabledCell$ = disabledCell.toSelector();
```

File: src/table/__integ__/inline-editing.test.ts [hunk 2]
```diff
@@ -150,3 +154,36 @@ test(
...
+  setupTest(async page => {
+    // Click on cell with disabled inline edit
+    await page.click(disabledCell$);
+
+    await expect(page.getText(disabledCellLiveRegion$)).resolves.toContain(
+      "You don't have the necessary permissions to change a BrowserStack origin."
+    );
+
+    // Dismiss with Escape
+    await page.keys(['Escape']);
```
Comment (abdhalees) [inline src/table/__integ__/inline-editing.test.ts:169]: when testing in Chrome, if I click on the disabled cell by mouse and then press `Escape` the popover doesn't get dismissed

it works only if I clicked on the lock icon, is that expected?

File: src/table/body-cell/disabled-inline-editor.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        {descriptionEl}
+        {isEditing && (
+          <PopoverContainer
+            size="medium"
+            fixedWidth={false}
+            position="top"
+            trackRef={iconRef}
+            arrow={position => <Arrow position={position} />}
+            renderWithPortal={true}
+            zIndex={7000}
```
Comment (abdhalees) [inline src/table/body-cell/disabled-inline-editor.tsx:99]: is there a reason it has this high number?

File: src/table/__integ__/inline-editing.test.ts (hunk 2 above)
Comment (timogasda) [reply src/table/__integ__/inline-editing.test.ts:169]: That's not expected. Will take a look

Comment (timogasda) [reply src/table/__integ__/inline-editing.test.ts:169]: @abdhalees I couldnt reproduce this 🤔

File: src/table/body-cell/disabled-inline-editor.tsx (hunk 3 above)
Comment (timogasda) [reply src/table/body-cell/disabled-inline-editor.tsx]: Updated to be the regular popover z-index

File: src/table/__integ__/inline-editing.test.ts (hunk 2 above)
Comment (abdhalees) [reply src/table/__integ__/inline-editing.test.ts:169]: oh I was looking at the outdated `preview` link and didn't notice that there is a new `dev-pages` link.

all good :)

**Review by abdhalees — APPROVED** [review]
I don't mind overriding codecov since the missing lines are covered by integ tests, but that will just be a problem for the next person who works on this file

**Comment by timogasda** [comment]
> I don't mind overriding codecov since the missing lines are covered by integ tests, but that will just be a problem for the next person who works on this file

I'll try to get the last few percent somehow haha

File: style-dictionary/visual-refresh/colors.ts [hunk 4]
```diff
@@ -174,6 +174,7 @@ const tokens: StyleDictionary.ColorsDictionary = {
   colorTextControlDisabled: '{colorTextInteractiveDisabled}',
   colorTextCounter: { light: '{colorGrey550}', dark: '{colorGrey450}' },
   colorTextDisabled: { light: '{colorGrey400}', dark: '{colorGrey550}' },
+  colorTextDisabledInlineEdit: { light: '{colorGrey600}', dark: '{colorGrey400}' },
```
Comment (abdhalees) [inline style-dictionary/visual-refresh/colors.ts:177]: this change should include a snapshot update

**Review by abdhalees — CHANGES_REQUESTED** [review]

**Review by abdhalees — APPROVED** [review]

**Review by abdhalees — APPROVED** [review]

**Pull request merged**